client/value.go

```diff
@@ -80,3 +84,25 @@ func (api *GoShimmerAPI) SendTransactionByJSON(txn webapi_sendtxnbyjson.Request)
 
 	return res.TransactionID, nil
 }
+
+// GetAllowedManaPledgeNodeIDs returns the list of allowed mana pledge IDs.
+func (api *GoShimmerAPI) GetAllowedManaPledgeNodeIDs() (map[mana.Type][]string, error) {
+	res := &webapi_allowedmanapledge.Response{}
+	if err := api.do(http.MethodGet, routeAllowedPledgeNodeIDs, nil, res); err != nil {
+		return nil, err
+	}
+
+	if len(res.Access.Allowed) == 0 {
+		return nil, errors.New("empty list of allowed access mana pledge IDs returned")
+	}
+	if len(res.Consensus.Allowed) == 0 {
+		return nil, errors.New("empty list of allowed consensus mana pledge IDs returned")
+	}
```

Comment:
Why is it an error if no pledgeNodeIDs are allowed?

Reply:
You are right, webapi endpoint should return error in this case.